src/deepsparse/pipeline_loggers/manager.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A manager that oversees all the pipeline loggers
+"""
+
+from typing import Any, Dict, List, Set, Union
+
+from deepsparse import PipelineLogger
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import (
+    IDENTIFIER as PROMETHEUS_IDENTIFIER,
+)
+from deepsparse.pipeline_loggers.prometheus_pipeline_logger import PrometheusLogger
+from deepsparse.timing.timing_schema import InferenceTimingSchema
+
+
+__all__ = ["LoggerManager"]
+
+SUPPORTED_LOGGERS = {PROMETHEUS_IDENTIFIER}
+
+
+class LoggerManager:
+    """
+    Object that contains multiple loggers for
+    the given inference pipeline.
+
+    The envisioned lifecycle of a logger
+    manager:
+
+    ```
+    pipeline = ... # define a pipeline
+    pipeline_name = pipeline.name # fetch the pipeline name
+
+    # create a LoggerManager
+    logger_manager = LoggerManager("logger_name")
+    or
+    logger_manager = LoggerManager(["logger_name_1", "logger_name_2", ...])
+
+    # log the data for the particular inference pipeline
+
+    timings, data = ...
+    logger.log_timings(pipeline_name, timings)
+    logger.log_data(pipeline_name, data)
+
+    ```
+
+    :param logger_identifiers: The identifier(s) of the logger types that
+        will be created within the scope of the manager. This can be either a
+        single identifier, or a list of identifiers (for multiple loggers).
+    :param supported_loggers: A set of supported logger identifiers; listed in
+        `SUPPORTED_LOGGERS` variable
+    """
+
+    def __init__(
+        self,
+        logger_identifiers: Union[str, List[str]],
```

Comment:
I'm going to recommend a larger change that is to just pass in the list of PipelineLoggers directly here. That will do two things:
1. You can probably just remove the identifier all together in that class (which would remove a big portion of this)
2. Make it more clear that LoggerManager is a "meta" PipelineLogger that is built up of a list of sub pipeline loggers.

Reply:
I like the idea, but I wonder how this would work, given that we use human-readable (writable) config to create the loggers. What I am saying is, that it would be natural to pass identifiers (string), directly parsed from the config, rather then logger classes, to create the manager.

Reply:
I think in that case the server should do the mapping from identifier or name to the class and construct the classes. This keeps the human readable stuff isolated into server, so it doesn't infect this code.

As far as how the server does the mapping it could either be strings like this, or we could even have users use the class name and then do a `getattr(deepsparse.pipeline_loggers, name)`, where name could be `PrometheusLogger`.